core/src/main/java/feast/core/service/JobService.java

```diff
@@ -0,0 +1,228 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package feast.core.service;
+
+import com.google.protobuf.InvalidProtocolBufferException;
+import feast.core.CoreServiceProto.ListFeatureSetsResponse;
+import feast.core.CoreServiceProto.ListIngestionJobsRequest;
+import feast.core.CoreServiceProto.ListIngestionJobsResponse;
+import feast.core.CoreServiceProto.RestartIngestionJobRequest;
+import feast.core.CoreServiceProto.RestartIngestionJobResponse;
+import feast.core.CoreServiceProto.StopIngestionJobRequest;
+import feast.core.CoreServiceProto.StopIngestionJobResponse;
+import feast.core.FeatureSetReferenceProto.FeatureSetReference;
+import feast.core.IngestionJobProto;
+import feast.core.dao.JobRepository;
+import feast.core.job.JobManager;
+import feast.core.model.FeatureSet;
+import feast.core.model.Job;
+import feast.core.model.JobStatus;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.NoSuchElementException;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+import javax.transaction.Transactional;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+/** Defines a Job Managemenent Service that allows users to manage feast ingestion jobs. */
+@Service
+public class JobService {
+  private JobRepository jobRepository;
+  private SpecService specService;
+  private Map<String, JobManager> jobManagers;
+
+  @Autowired
+  public JobService(
+      JobRepository jobRepository, SpecService specService, List<JobManager> jobManagerList) {
+    this.jobRepository = jobRepository;
+    this.specService = specService;
+
+    this.jobManagers = new HashMap<>();
+    for (JobManager manager : jobManagerList) {
+      this.jobManagers.put(manager.getRunnerType().getName(), manager);
+    }
+  }
+
+  /* Job Service API */
+  /**
+   * List Ingestion Jobs in feast matching the given request
+   *
+   * @param request list ingestion jobs request specifying which jobs to include
+   * @throws InvalidArgumentException when given filter in a unsupported configuration
+   * @throws InvalidProtocolBufferException on error when constructing response protobuf
+   * @return list ingestion jobs response
+   */
+  public ListIngestionJobsResponse listJobs(ListIngestionJobsRequest request)
```

Comment:
Since this method requires multiple separate calls to the db, it should be annotated with 
`@Transactional(readOnly=True)` to ensure that you're reading from the same snapshot of data